### PR TITLE
University Can Buy a Subscription

### PR DESCRIPTION
**Pull request by sethnejame**
This PR does the following:

- Adds the subscriber column for the user model for paid/unpaid universities
- Builds the subscription controller/routes that will receive Stripe payments from the front end
- Tests both the user model for the subscriber column and the subscription controller/routes that handle Stripe payments

```
As a University,
In order to gain access to the website services,
I would like to be able to buy a subscription.
```

Pivotal Tracker link: https://www.pivotaltracker.com/story/show/167890672

### CODE REVIEW

**Comment by faraznaeem** [comment]
What is the status of this one?

**Comment by sethnejame** [comment]
> What is the status of this one?

We've barely got started. . .we ended up getting derailed today fixing application bugs (we deployed this and found a few problems with the frontend = backend talking) and trying to get the navbar changes merged into the project.  Going to continue on this now.

File: spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      expect(response_json['message']).to eq 'Payment successful'
+    end
+
+    it 'requires a charge amount' do
+      expect {
+        charge =
+          Stripe::Charge.create(
+            currency: 'sek', customer: university, stripeToken: :stripeToken
+          )
+      }.to raise_error(Stripe::InvalidRequestError)
```
Comment (faraznaeem) [inline spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb]: Why are you raising the `Stripe::InvalidRequestError` ?
While in the happy path you are raising ` 'Payment successful'
`

File: spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    end
+
+    it 'requires a valid credit card' do
+      expect {
+        charge =
+          Stripe::Charge.create(
+            currency: 'sek', stripeToken: 'whatever dude', amount: 10_000
+          )
+      }.to raise_error(Stripe::InvalidRequestError)
```
Comment (faraznaeem) [inline spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb]: Same here?

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -29,53 +29,81 @@
...
 
       emails.each do |email|
         it { is_expected.to allow_value(email).for(:email) }
       end
     end
 
     describe 'User can have a University Role' do
-      let(:user){create :user, email: 'oxford@oxford.edu', role: :university}
-     
+      let(:user) { create :user, email: 'oxford@oxford.edu', role: :university }
```
Comment (faraznaeem) [inline spec/models/user_spec.rb:68]: Why are you guys are not using `FactoryBot`

File: spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+RSpec.describe Api::V0::SubscriptionsController, type: :request do
+  describe 'POST /api/v0/subscriptions' do
+    let(:university) { create(:user, role: :university) }
```
Comment (faraznaeem) [inline spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb]: `FactoryBot`?

**Review by faraznaeem — DISMISSED** [review]

**Review by oliverochman — APPROVED** [review]
LGTM!

File: spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    it 'return a 200 status' do
+      expect(response.status).to eq 200
+    end
+
+    it 'returns a success message' do
+      expect(response_json['message']).to eq 'Payment successful'
+    end
+  end
+
+  describe 'Payment does not post successfully (Sad Path)' do
```
Comment (tochman) [inline spec/requests/api/v0/uni_can_pay_for_subscription_spec.rb]: Okay, so these tests are in a way relevant, but they are NOT request specs, and should not be in this file. If you are interested I can elaborate on the subject in a session. If you want to test the sad path, you should make a post request without a `stripeToken` param.

**Review by tochman — DISMISSED** [review]
Please see my comment about the specs for unsuccessful payments.

**Pull request merged**